tcl/db.tcl

```diff
@@ -387,7 +387,13 @@ proc qc::db_trans {args} {
 	    } else {
 		incr db_trans_level($db) -1
 	    }
-	    return -code $code $result
+            
+            if { $return_code == 2 && [dict get $options -code] == 0 } {
+                dict set options -code return
+            } else {
+                dict incr options -level
```

Comment:
I've added this increment to level because I realised that under most circumstances, since we're executing code in the caller's stack frame, then we want to return values while taking into account the caller's stack frame.

I'll add another comment shortly with the testing performed and attempt to explain why the level should be incremented here.